src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Base64Validator.cs

```diff
@@ -0,0 +1,201 @@
+                    {
+                        // An invalid char was encountered.
+                        decodedLength = 0;
+                        return false;
+                    }
+
+                    if (indexOfPaddingInvalidOrWhitespace == base64Text.Length - 1)
+                    {
+                        // The end of the input has been reached.
+                        break;
+                    }
+
+                    // If no padding is found, slice and use IndexOfAnyExcept to look for the next invalid char.
+                    if (paddingCount == 0)
+                    {
+                        indexOfPaddingInvalidOrWhitespace = base64Text
+                            .Slice(indexOfPaddingInvalidOrWhitespace + 1, base64Text.Length - indexOfPaddingInvalidOrWhitespace - 1)
+                            .IndexOfAnyExcept(validBase64Chars)
+                            + indexOfPaddingInvalidOrWhitespace + 1; // Add current index offset.
+                    }
+                    // If padding is already found, simply increment, as the common case might have 2 sequential padding chars.
+                    else
+                    {
+                        indexOfPaddingInvalidOrWhitespace++;
+                    }
+                }
+
+                // If the invalid chars all consisted of whitespace, the input will be empty.
+                if (length == 0)
+                {
+                    decodedLength = 0;
+                    return true;
+                }
+            }
+
+            if (length % 4 != 0)
+            {
+                decodedLength = 0;
+                return false;
+            }
+
+            // Remove padding to get exact length
+            decodedLength = (int)((uint)length / 4 * 3) - paddingCount;
+            return true;
+        }
+
+        /// <summary>
+        /// Validates the span of UTF-8 encoded text represented as base64 into binary data.
+        /// </summary>
+        /// <param name="base64TextUtf8">The input span which contains UTF-8 encoded text in base64 that needs to be validated.</param>
+        /// <returns>true if <paramref name="base64TextUtf8"/> is decodable; otherwise, false.</returns>
+        public static bool IsValid(ReadOnlySpan<byte> base64TextUtf8) => IsValid(base64TextUtf8, out int _);
+
+        /// <summary>
+        /// Validates the span of UTF-8 encoded text represented as base64 into binary data.
+        /// </summary>
+        /// <param name="base64TextUtf8">The input span which contains UTF-8 encoded text in base64 that needs to be validated.</param>
+        /// <param name="decodedLength">The maximum length (in bytes) if you were to decode the base 64 encoded text <paramref name="base64TextUtf8"/> within a byte span.</param>
+        /// <returns>true if <paramref name="base64TextUtf8"/> is decodable; otherwise, false.</returns>
+        public static bool IsValid(ReadOnlySpan<byte> base64TextUtf8, out int decodedLength)
```

Comment:
The implementation of `byte` and `char` is the same. This duplication can be avoided by using static abstract interface members (with internal / private struct types provide the actual implementation where the difference between byte and char exists -- struct to allow generic specialization, thus better codegen and may inline).